packages/p2p-v2/src/pages/orders/components/OrderDetailsCard/OrderDetailsCardFooter/OrderDetailsCardFooter.tsx

```diff
@@ -7,57 +8,86 @@ import './OrderDetailsCardFooter.scss';
-        );
-
-    if (shouldShowOnlyComplainButton)
-        return (
-            <div className='p2p-v2-order-details-card-footer justify-end'>
-                <Button className='border-2' color='primary-light' size='lg' textSize={textSize} variant='ghost'>
-                    Complain
-                </Button>
-            </div>
-        );
-
-    if (shouldShowOnlyReceivedButton)
-        return (
-            <div className='p2p-v2-order-details-card-footer justify-end'>
-                <Button size='lg' textSize={textSize}>
-                    I’ve received payment
-                </Button>
-            </div>
-        );
-
-    return null;
+    return (
+        <div className='p2p-v2-order-details-card-footer'>
+            {shouldShowCancelAndPaidButton && (
+                <div className='gap-3 ml-auto'>
+                    <Button className='border-2' color='black' size='lg' textSize={textSize} variant='outlined'>
+                        Cancel order
+                    </Button>
+                    <Button size='lg' textSize={textSize}>
+                        I’ve paid
+                    </Button>
+                </div>
+            )}
+            {shouldShowComplainAndReceivedButton && (
+                <div className='justify-between'>
+                    <Button
+                        className='border-2'
+                        color='primary-light'
+                        onClick={() => setIsModalOpen(true)}
+                        size='lg'
+                        textSize={textSize}
+                        variant='ghost'
+                    >
+                        Complain
+                    </Button>
+                    <Button size='lg' textSize={textSize}>
+                        I’ve received payment
+                    </Button>
+                </div>
+            )}
+            {shouldShowOnlyComplainButton && (
+                <div className='ml-auto'>
+                    <Button
+                        className='border-2'
+                        color='primary-light'
+                        onClick={() => setIsModalOpen(true)}
+                        size='lg'
+                        textSize={textSize}
+                        variant='ghost'
+                    >
+                        Complain
+                    </Button>
+                </div>
+            )}
+            {shouldShowOnlyReceivedButton && (
+                <div className='ml-auto'>
+                    <Button size='lg' textSize={textSize}>
+                        I’ve received payment
+                    </Button>
+                </div>
+            )}
+            {isModalOpen && (
+                <OrderDetailsComplainModal
+                    id={id}
+                    isBuyOrderForUser={isBuyOrderForUser}
+                    isModalOpen={isModalOpen}
+                    onRequestClose={() => setIsModalOpen(false)}
+                />
+            )}
+        </div>
+    );
```

Comment:
why are we combining all the different buttons together under one return? The `OrderDetailsComplainModal` should only be shown for shouldShowComplainAndReceiveButton right? Better we keep it as before

so you just need to do this:

```
if (shouldShowComplainAndReceivedButton)
        return (
            <>
                  <div className='p2p-v2-order-details-card-footer justify-between'>
                      <Button className='border-2' color='primary-light' size='lg' textSize={textSize} variant='ghost'>
                          Complain
                      </Button>
                      <Button size='lg' textSize={textSize}>
                          I’ve received payment
                      </Button>
                  </div>
                  <OrderDetailsComplainModal
                      id={id}
                      isBuyOrderForUser={isBuyOrderForUser}
                      isModalOpen={isModalOpen}
                      onRequestClose={() => setIsModalOpen(false)}
                  />
            </>
        );

Reply:
also regarding how to implement the modal, I think its a little different right @adrienne-deriv

Reply:
there are 2 conditions with complain buttons, so adding it at both the places will be kind of repetitive(which will be true for i've received payment button as well). its better to avoid repetitive code. also only one condition will be true at one time, so i dont think this would cause issue.
regarding the modal implementation there are quite a few issues with the modal manager. so i reverted the changes and implemented it the old way itself.

Reply:
for the modal manager implementation best you liase with @adrienne-deriv to try and see if there are any kinks or improvements to how we can improve it, lets try not to go the old way since modal manager available

Reply:
You can implement without the conditional rendering like:
```
const { modal, showModal, hideModal } = useModalManager()

const toShowModalFn = () => {
    showModal('OrderDetailsComplainModal')
}

<OrderDetailsComplainModal
                    id={id}
                    isBuyOrderForUser={isBuyOrderForUser}
                    isModalOpen={modal.OrderDetailsComplainModal}
                    onRequestClose={() => hideModal()}
                />
```

Reply:
Oh wait `modal` is not being returned, could you try returning it from the hook, like:
```
   return {
       modal: isModalOpenScopes,
       ...
   }
```

Reply:
If it doesn't work, just use `isModalOpenFor`:
```
const { isModalOpenForm, showModal, hideModal } = useModalManager() 
const toShowModalFn = () => { showModal('OrderDetailsComplainModal') } 

return <OrderDetailsComplainModal 
                id={id} isBuyOrderForUser={isBuyOrderForUser} 
                isModalOpen={isModalOpenFor('OrderDetailsComplainModal')} 
               onRequestClose={() => hideModal()} />
```